src/modules/keyboardmanager/ui/ShortcutControl.cpp

```diff
@@ -55,6 +55,237 @@ void ShortcutControl::AddNewShortcutControlRow(StackPanel& parent, Shortcut orig
     parent.Children().Append(tableRow);
 }
 
+// Function to add a drop down to the shortcut stack panel
+ComboBox ShortcutControl::AddDropDown(StackPanel parent, const int& rowIndex, const int& colIndex)
+{
+    ComboBox shortcutDropDown;
+    shortcutDropDown.Width(100);
+    shortcutDropDown.MaxDropDownHeight(200);
+    shortcutDropDown.ItemsSource(keyboardManagerState->keyboardMap.GetKeyList(true).first);
+    // Flyout to display the warning on the drop down element
+    Flyout warningFlyout;
+    TextBlock warningMessage;
+    warningFlyout.Content(warningMessage);
+    shortcutDropDown.ContextFlyout().SetAttachedFlyout((FrameworkElement)shortcutDropDown, warningFlyout);
+
+    // drop down selection handler
+    shortcutDropDown.SelectionChanged([&, rowIndex, colIndex, parent, warningMessage](IInspectable const& sender, SelectionChangedEventArgs const&) {

```

Comment:
Maybe we should add a Handler to the `DropDownOpened` event, where the current Keyboard Layout is checked and the keys shown are updated accordingly

Reply:
Done. There was also the case where when you use "Type key" the corresponding drop down has to be set, and if there was a language change while on the type key window it would not be reflected on the drop down since it wasn't opened yet. This was fixed by resetting the drop down item source whenever the user types in a key, that way the latest language is used and it does not happen too often.